gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -565,6 +570,64 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
 
     return new FSDataInputStream(in);
   }
+  // Conf not used in the method for sometime, as some changes need to be made for laters
+  private void initThreadPools() {
+    int maxThreads = DEFAULT_MAX_THREADS.getDefault();
+    if (maxThreads < 2) {
+      logger.atWarning().log("Maximum number of threads should at least be 2.");
+      maxThreads = 2;
+    }
+    int keepAliveTime = KEEP_ALIVE_TIME.getDefault();
+    unboundedThreadPool =
+        new ThreadPoolExecutor(
+            maxThreads,
+            Integer.MAX_VALUE,
+            keepAliveTime,
+            TimeUnit.SECONDS,
+            new LinkedBlockingQueue<>(),
+            BlockingThreadPoolExecutorService.newDaemonThreadFactory("gcs-transfer-unbounded"));
+    unboundedThreadPool.allowCoreThreadTimeOut(true);
+  }
+
+  @Override
+  public CompletableFuture<FSDataInputStream> openFileWithOptions(
+      final Path rawPath, final OpenFileParameters parameters) throws IOException {
+    final Path path = makeQualified(rawPath);
+    Set<String> mandatoryKeys = parameters.getMandatoryKeys();
+    AbstractFSBuilderImpl.rejectUnknownMandatoryKeys(
+        mandatoryKeys, Collections.emptySet(), "for " + path);
+    StorageResourceId storageResourceId = StorageResourceId.fromUriPath(rawPath.toUri(), false);
+    GoogleCloudStorageItemInfo itemInfo = this.getGcsFs().getGcs().getItemInfo(storageResourceId);
```

Comment:
Retrieving item info issues GCS request and defeats the purpose of the `openFileWithOptions` which will have status in the parameters already. Instead we should use status from the parameters:
```java
FileStatus status = parameters.getStatus();
```

Reply:
@medb parametersgetStatus() is empty/null

Reply:
It will be not empty if caller will call it with `parameters` that has it:

```java
new OpenFileParameters().withStatus(status);
```

If `status` is not in the parameters then we may get info from GCS, but the main use case for this method is when `status` is passed via `parameters`.